src/test/statistics/chunk_statistics/range_filter_test.cpp

```diff
@@ -125,28 +70,69 @@ TYPED_TEST(RangeFilterTest, SingleRange) {
 
   // cannot prune values in between, even though non-existent
-  EXPECT_FALSE(filter->can_prune(PredicateCondition::Equals, {this->_in_between}));
+  EXPECT_FALSE(filter->can_prune(PredicateCondition::Equals, TypeParam{this->_value_in_gap}));
 
-  EXPECT_FALSE(filter->can_prune(PredicateCondition::LessThanEquals, {this->_max_value}));
-  EXPECT_TRUE(filter->can_prune(PredicateCondition::GreaterThan, {this->_max_value}));
+  EXPECT_FALSE(filter->can_prune(PredicateCondition::LessThanEquals, TypeParam{this->_max_value}));
+  EXPECT_TRUE(filter->can_prune(PredicateCondition::GreaterThan, TypeParam{this->_max_value}));
+
+  EXPECT_TRUE(filter->can_prune(PredicateCondition::Between, TypeParam{-3000}, TypeParam{-2000}));
 }
 
 // create range filters with varying number of ranges/gaps
 TYPED_TEST(RangeFilterTest, MultipleRanges) {
-  for (auto gap_count = size_t{0}; gap_count < this->_values.size() * 2; ++gap_count) {
-    const auto filter = this->test_varying_range_filter_size(gap_count, this->_values);
+  const auto first_gap_min = TypeParam{104};
+  const auto first_gap_max = TypeParam{123455};
+
+  const auto second_gap_min = TypeParam{-999};
+  const auto second_gap_max = TypeParam{1};
 
-    // _in_between should always prune if we have more than one range
-    if (gap_count > 1) {
-      EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, this->_in_between));
+  const auto third_gap_min = TypeParam{18};
+  const auto third_gap_max = TypeParam{99};
+
+  {
+    // Throw when range filter shall include 0 range values.
+    EXPECT_THROW((RangeFilter<TypeParam>::build_filter(this->_values, 0)), std::logic_error);
+  }
+  {
+    const auto filter = RangeFilter<TypeParam>::build_filter(this->_values, 2);
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, this->_value_in_gap));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, first_gap_min));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Between, first_gap_min, first_gap_max));
+
+    EXPECT_FALSE(filter->can_prune(PredicateCondition::Between, second_gap_min, second_gap_max));
+    EXPECT_FALSE(filter->can_prune(PredicateCondition::Between, third_gap_min, third_gap_max));
+  }
+  {
+    const auto filter = RangeFilter<TypeParam>::build_filter(this->_values, 3);
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, this->_value_in_gap));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, first_gap_min));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Between, first_gap_min, first_gap_max));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, second_gap_min));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Between, second_gap_min, second_gap_max));
+
+    EXPECT_FALSE(filter->can_prune(PredicateCondition::Between, third_gap_min, third_gap_max));
+  }
+  {
+    // starting with 4 ranges, all tested gaps should be covered
+    for (uint32_t range_count : {4, 5, 100, 1'000}) {
```

Comment:
just use auto here? I feel like sooner or later a compiler will start complaining about "assigning signed to non signed int literal"